### PR TITLE
[Minor] Improve error message when bitwise_* operator takes wrong unsupported type

### LINKED ISSUE #11249: Make error message better when `bitwise_*` operator takes wrong argument type
### Describe the bug

This is not a bug, but the error message can be better:
The following queries' problem is just `bitwise_*` operator takes the not-supported argument type, so it should be a planning error.
Current `Internal error` messages can be confusing, or cause a false positive for testing code.

```shell
> select 2.0 << 1.5;
Internal error: Data type Float64 not supported for binary operation 'bitwise_shift_left' on dyn arrays.
This was likely caused by a bug in DataFusion's code and we would welcome that you file an bug report in our issue tracker
> select 3.14 >> 1.2;
Internal error: Data type Float64 not supported for binary operation 'bitwise_shift_right' on dyn arrays.
This was likely caused by a bug in DataFusion's code and we would welcome that you file an bug report in our issue tracker
> select 3.14 | 1.3;
Internal error: Data type Float64 not supported for binary operation 'bitwise_or' on dyn arrays.
This was likely caused by a bug in DataFusion's code and we would welcome that you file an bug report in our issue tracker
> select 1.3 ^ 3.14;
Internal error: Data type Float64 not supported for binary operation 'bitwise_xor' on dyn arrays.
This was likely caused by a bug in DataFusion's code and we would welcome that you file an bug report in our issue tracker
> select 1.5 & 1.6;
Internal error: Data type Float64 not supported for binary operation 'bitwise_and' on dyn arrays.
This was likely caused by a bug in DataFusion's code and we would welcome that you file an bug report in our issue tracker
```

### To Reproduce

_No response_

### Expected behavior

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by dharanad**
## Which issue does this PR close?

Closes #11249

### CODE REVIEW

File: datafusion/physical-expr/src/expressions/binary/kernels.rs [hunk 1]
```diff
@@ -24,7 +24,7 @@ use arrow::compute::kernels::bitwise::{
     bitwise_xor, bitwise_xor_scalar,
 };
 use arrow::datatypes::DataType;
-use datafusion_common::internal_err;
+use datafusion_common::plan_err;
```
Comment (alamb) [inline datafusion/physical-expr/src/expressions/binary/kernels.rs:27]: Perhaps `not_impl_err` would be more appropriate?

https://docs.rs/datafusion/latest/datafusion/common/macro.not_impl_err.html

**Review by alamb — APPROVED** [review]
Thank you @dharanad  -- this looks much better than what is currently done. It might be better to use `not_impl_err` but we could always do that as a follow on PR or never too

Thanks again!

File: datafusion/physical-expr/src/expressions/binary/kernels.rs (hunk 1 above)
Comment (dharanad) [reply datafusion/physical-expr/src/expressions/binary/kernels.rs:27]: it seems unlikely that we would support such an operation so `plan_err` should be good right ? However, I may be overlooking some factors. Could you please clarify if there's a specific rationale behind this suggestion that I might have missed?

Comment (alamb) [reply datafusion/physical-expr/src/expressions/binary/kernels.rs:27]: I was thinking that most `plan_err` s are issues with a user's query (some bug they should fix) where as `not_impl` signals that there isn't some bug with the query

This is kind of documented here: 
* https://docs.rs/datafusion/latest/datafusion/common/enum.DataFusionError.html#variant.Plan
* https://docs.rs/datafusion/latest/datafusion/common/enum.DataFusionError.html#variant.NotImplemented

**Comment by alamb** [comment]
Plan error is certainly better than internal error. We can consider switching to not implemented error as a follow on PR

Thanks @dharanad

File: datafusion/physical-expr/src/expressions/binary/kernels.rs (hunk 1 above)
Comment (dharanad) [reply datafusion/physical-expr/src/expressions/binary/kernels.rs:27]: Thats clear. Thank You

**Pull request merged**